src/Stack/Package.hs

```diff
@@ -129,14 +127,14 @@ rawParseGPD
   -> BS.ByteString
   -> m ([PWarning], GenericPackageDescription)
 rawParseGPD key bs =
-    case parseGenericPackageDescription chars of
-       ParseFailed e -> throwM $ PackageInvalidCabalFile key e
-       ParseOk warnings gpkg -> return (warnings,gpkg)
+    case eres of
+      Left (mversion, errs) -> throwM $ PackageInvalidCabalFile key
+        (fromCabalVersion <$> mversion)
+        errs
+        warnings
+      Right gpkg -> return (warnings, gpkg)
   where
-    chars = T.unpack (dropBOM (decodeUtf8With lenientDecode bs))
-
-    -- https://github.com/haskell/hackage-server/issues/351
-    dropBOM t = fromMaybe t $ T.stripPrefix "\xFEFF" t
```

Comment:
@phadej Can we rely on Cabal stripping out the byte-order marker from the bytestring, or do we need to do something more intelligent here?

Reply:
Cabal strips BOM (and warns about, if sees it). `parseGenericPackageDescription` also verifies that `ByteString` looks like valid UTF8, so you don't to do that either.

TL;DR you don't need to do any preprocessing.

Reply:
Awesome, thanks!